test/SlowTests/Issues/RavenDB-17937.cs

```diff
@@ -0,0 +1,142 @@
+using System.Linq;
+using System.Threading.Tasks;
+using FastTests;
+using FastTests.Client;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Operations.Indexes;
+using Raven.Client.Exceptions;
+using Raven.Client.ServerWide;
+using Raven.Client.ServerWide.Operations;
+using Raven.Server.Config;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_17937 : RavenTestBase
+    {
+        public RavenDB_17937(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Theory]
+        [InlineData("Index")]
+        [InlineData("ZIndex")]
+        public async Task Can_Add_Database_Folder_With_Side_By_Side_Indexes(string indexName)
+        {
+            var path = NewDataPath();
+
+            using (var store = GetDocumentStore(new Options
+                   {
+                       RunInMemory = false,
+                       Path = path
+            }))
+            {
+                using (var session = store.OpenAsyncSession())
+                {
+                    await session.StoreAsync(new Query.Order());
+                    await session.SaveChangesAsync();
+                }
+
+                var index1 = new Index1(indexName);
+                await index1.ExecuteAsync(store);
+                await new Index2(indexName).ExecuteAsync(store);
+
+                WaitForIndexing(store, allowErrors: true, timeout: TimeSpan.FromSeconds(5));
+
+                var record = await store.Maintenance.Server.SendAsync(new GetDatabaseRecordOperation(store.Database));
+                Assert.Equal(1, record.Indexes.Count);
+                Assert.Equal(index1.IndexName, record.Indexes.First().Key);
+
+                var indexStats = await store.Maintenance.SendAsync(new GetIndexesStatisticsOperation());
+                Assert.Equal(2, indexStats.Length);
+
+                await store.Maintenance.Server.SendAsync(new DeleteDatabasesOperation(store.Database, hardDelete: false));
+
+                await store.Maintenance.Server.SendAsync(new CreateDatabaseOperation(new DatabaseRecord(store.Database)
+                {
+                    Settings =
+                    {
+                        {RavenConfiguration.GetKey(x => x.Core.RunInMemory), "false" },
+                        {RavenConfiguration.GetKey(x => x.Core.DataDirectory), path }
+                    }
+                }));
+
+                record = await store.Maintenance.Server.SendAsync(new GetDatabaseRecordOperation(store.Database));
+                Assert.Equal(index1.IndexName, record.Indexes.First().Key);
+                Assert.Equal("docs.Orders.Select(order => new {\r\n    order = order,\r\n    x = 0\r\n}).Select(this0 => new {\r\n    Count = 1 / this0.x\r\n})", record.Indexes.First().Value.Maps.First());
+
+                indexStats = await store.Maintenance.SendAsync(new GetIndexesStatisticsOperation());
+                Assert.Equal(2, indexStats.Length);
+            }
```

Comment:
@grisha-kotler I wonder if we shouldn't change the definition at this point - to see that it works correctly. There won't be any clash in the paths with the new replacement index?